e2e/helpers/tables.ts

```diff
@@ -0,0 +1,420 @@
+          const arrowSelector =
+            decadeDifference > 0 ? '.prev:visible' : '.next:visible'
+          for (let i = 0; i < decadeDifference; i++) {
+            await page.locator(arrowSelector).click()
+          }
+        })
+      }
+
+      await page
+        .getByText(year.toString(), { exact: true })
+        .locator('visible=true')
+        .click()
+    })
+
+    await test.step('select month', async () => {
+      await page
+        .getByText(month, { exact: true })
+        .locator('visible=true')
+        .click()
+    })
+
+    await test.step('select day', async () => {
+      let dayLocator = page.getByRole('cell', {
+        name: day.toString(),
+        exact: true,
+      })
+
+      // handle case where day may also be visible in previous or next month
+      // e.g. 1 or 30
+      if ((await dayLocator.count()) > 1) {
+        dayLocator = day > 15 ? dayLocator.nth(1) : dayLocator.first()
+      }
+      await dayLocator.click()
+    })
+  })
+}
+
+export const selectTableRestrictValues = async (
+  cell: Locator,
+  option: string,
+) => {
+  await cell.getByRole('combobox').selectOption(option)
+}
+
+export const enterTableListValues = async (
+  page: Page,
+  cell: Locator,
+  values: string | number | string[],
+) => {
+  await test.step('enter table list values', async () => {
+    const modalHeader = await test.step('open edit modal', async () => {
+      await cell.click()
+      const modalHeader = page.getByRole('heading', {
+        name: 'Edit Values',
+      })
+      await expect(modalHeader).toBeVisible()
+      return modalHeader
+    })
+
+    await test.step('add values', async () => {
+      await page.getByRole('button', { name: 'PASTE NEW VALUES' }).click()
+      await page
+        .getByRole('textbox', {
+          name: 'Paste comma or tab delimited values here.',
+        })
+        .fill(Array.isArray(values) ? values.join(',') : values.toString())
+      await page.getByRole('button', { name: 'ADD', exact: true }).click()
+    })
+
+    await test.step('save changes', async () => {
+      await page.getByRole('button', { name: 'OK' }).click()
+      await expect(modalHeader).not.toBeVisible()
+    })
+  })
+}
+
+export const saveTableDataChanges = async (page: Page) => {
+  await test.step('save table data changes', async () => {
+    await page.getByRole('button', { name: 'Save' }).click()
+    await expect(getTableEditorHeader(page)).not.toBeVisible({
+      timeout: defaultExpectTimeout * 3, // allow time for changes to be applied
+    })
+    await expect(page.getByText('Applying changes...')).not.toBeVisible()
+  })
+}
+
+// Validate table data ---------------------------------------------------------
+const getTableRows = async (page: Page) => {
+  return page.getByTestId('SynapseTable').getByRole('row')
+}
+
+export const expectTableColumnHeaders = async (
+  tableRows: Locator,
+  columnsSchemaConfig: ColumnSchemaConfig[],
+  columnIndexOffset: number,
```

Comment:
Provide a `columnIndexOffset`, so this function can be used to check the table column headers when entering data into the table (and there is a column of checkboxes, so offset is 1) and when viewing the table (and the offset is 0).